### PR TITLE
🐛🎨External Clusters: improve work stealing

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
This PR aims to improve the situation with external clusters that do not optimally run tasks

- 🐛 fix an issue where the computational autoscaling could not scale down due to missing docker node labels
- 🐛 fix an issue where sending 8 jobs, then 6 jobs would not scale up the cluster to 10 if 10 is the maximum allowed number of machines
- 🎨 improve work stealing, by allowing the clusters-keeper to change the ```worker-saturation``` value, which is now set by default to ```inf``` which means the worker balancing is scheduler side (whatever that means). the default value of ```1.1``` seemed to not work well, ```1.0``` also not. This seemed like a good compromise although I don't find that justified from the detailed references below. Therefore this should be tested.
- 🎨 Allow to have settings that can be a union of a type and a ```typing.Literal``` (@pcrespov please double check that part and let me know what needs to be checked, I am not completely sure of the fix)
- ⚗️ Added ```CLUSTERS_KEEPER_DASK_WORKER_SATURATION``` environment variable to allow to "play" with the deployed dask-scheduler ```worker-saturation``` parameter.

[dask explanations about dask task scheduling concepts](https://distributed.dask.org/en/stable/scheduling-policies.html)
[very detailed explanations for dask work stealing concepts](https://selectfrom.dev/deep-dive-into-dask-distributed-scheduler-9fdb3b36b7c7)
## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))
- https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/475

### CODE REVIEW

**Comment by YuryHrytsuk** [comment]
@sanderegg i see new ENV. Is there a PR in `osparc-config`?

**Comment by sanderegg** [comment]
> @sanderegg i see new ENV. Is there a PR in `osparc-config`?

@YuryHrytsuk  I guess you are talking about this?
https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/475

**Comment by YuryHrytsuk** [comment]
> > @sanderegg i see new ENV. Is there a PR in `osparc-config`?
> 
> @YuryHrytsuk I guess you are talking about this? https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/475

Thank you

**Review by YuryHrytsuk — APPROVED** [review]

**Review by mrnicegyu11 — APPROVED** [review]
all good from the ops side of view

File: services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py [hunk 1]
```diff
@@ -280,6 +289,12 @@ class ApplicationSettings(BaseCustomSettings, MixinLoggingSettings):
         description="overrides the default number of threads in the dask-sidecars, setting it to 0 will use the default (see description in dask-sidecar)",
     )
 
+    CLUSTERS_KEEPER_DASK_WORKER_SATURATION: NonNegativeFloat | Literal["inf"] = Field(
+        default="inf",
+        description="override the dask scheduler 'worker-saturation' field"
+        ", see https://selectfrom.dev/deep-dive-into-dask-distributed-scheduler-9fdb3b36b7c7",
```
Comment (mguidon) [inline services/clusters-keeper/src/simcore_service_clusters_keeper/core/settings.py:295]: Interesting.

**Review by mguidon — APPROVED** [review]

**Review by matusdrobuliak66 — APPROVED** [review]

**Pull request merged**